pkg/kubectl/rolling_updater.go

```diff
@@ -488,25 +492,29 @@ func (r *RollingUpdater) cleanupWithClients(oldRc, newRc *api.ReplicationControl
 
 // func extractMaxValue is a helper to extract config max values as either
 // absolute numbers or based on percentages of the given value.
-func extractMaxValue(field intstr.IntOrString, name string, value int) (int, error) {
+func extractMaxValue(field intstr.IntOrString, name string, value int) (int, bool, error) {
 	switch field.Type {
 	case intstr.Int:
 		if field.IntVal < 0 {
-			return 0, fmt.Errorf("%s must be >= 0", name)
+			return 0, false, fmt.Errorf("%s must be >= 0", name)
 		}
-		return field.IntValue(), nil
+		return field.IntValue(), false, nil
 	case intstr.String:
 		s := strings.Replace(field.StrVal, "%", "", -1)
 		v, err := strconv.Atoi(s)
 		if err != nil {
-			return 0, fmt.Errorf("invalid %s value %q: %v", name, field.StrVal, err)
+			return 0, true, fmt.Errorf("invalid %s value %q: %v", name, field.StrVal, err)
 		}
 		if v < 0 {
-			return 0, fmt.Errorf("%s must be >= 0", name)
+			return 0, true, fmt.Errorf("%s must be >= 0", name)
 		}
-		return int(math.Ceil(float64(value) * (float64(v)) / 100)), nil
+		return int(math.Ceil(float64(value) * (float64(v)) / 100)), true, nil
 	}
-	return 0, fmt.Errorf("invalid kind %q for %s", field.Type, name)
+	return 0, false, fmt.Errorf("invalid kind %q for %s", field.Type, name)
+}
+
+func isOneHundredPercent(field intstr.IntOrString) bool {
+	return field.String() == "100%"
```

Comment:
If maxUavailable is not a percent and is equal to the replica count then there is nothing we need to do. We should make sure when it's a percent that if it's not 100% then we won't end up with zero minimum availability. I will remove this part of code anyway and switch math.Ceil to math.Floor in `extractMaxValue`.